### PR TITLE
Fix scanner_iter() db session out of sync when used by multiple simultaneous queries.

### LINKED ISSUE #1711: Scanner: ProgrammingError: (_mysql_exceptions.ProgrammingError) (2014, "Commands out of sync; you can't run this command now")
Scanners do runs and most of them complete just fine by populating the violation table, but a few of them failed with the same SQLalchemy error, example:

2018-06-25 13:43:58,974 ERROR [forseti-security] google.cloud.forseti.scanner.scanner(run): Error running scanner: BucketsAclScanner
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/forseti_security-2.0.0-py2.7.egg/google/cloud/forseti/scanner/scanner.py", line 112, in run
    scanner.run()
  File "/usr/local/lib/python2.7/dist-packages/forseti_security-2.0.0-py2.7.egg/google/cloud/forseti/scanner/scanners/bucket_rules_scanner.py", line 137, in run
    buckets_acls = self._retrieve()
  File "/usr/local/lib/python2.7/dist-packages/forseti_security-2.0.0-py2.7.egg/google/cloud/forseti/scanner/scanners/bucket_rules_scanner.py", line 125, in _retrieve
    project_id = bucket.parent.name
  File "build/bdist.linux-x86_64/egg/sqlalchemy/orm/attributes.py", line 242, in __get__
    return self.impl.get(instance_state(instance), dict_)
  File "build/bdist.linux-x86_64/egg/sqlalchemy/orm/attributes.py", line 599, in get
    value = self.callable_(state, passive)
  File "build/bdist.linux-x86_64/egg/sqlalchemy/orm/strategies.py", line 630, in _load_for_state
    session, state, primary_key_identity, passive)
  File "<string>", line 1, in <lambda>
  File "build/bdist.linux-x86_64/egg/sqlalchemy/orm/strategies.py", line 719, in _emit_lazyload
    session.query(self.mapper), primary_key_identity)
  File "build/bdist.linux-x86_64/egg/sqlalchemy/ext/baked.py", line 501, in _load_on_pk_identity
    result = list(bq.for_session(self.session).params(**params))
  File "build/bdist.linux-x86_64/egg/sqlalchemy/ext/baked.py", line 346, in __iter__
    return q._execute_and_instances(context)
  File "build/bdist.linux-x86_64/egg/sqlalchemy/orm/query.py", line 2958, in _execute_and_instances
    result = conn.execute(querycontext.statement, self._params)
  File "build/bdist.linux-x86_64/egg/sqlalchemy/engine/base.py", line 948, in execute
    return meth(self, multiparams, params)
  File "build/bdist.linux-x86_64/egg/sqlalchemy/sql/elements.py", line 269, in _execute_on_connection
    return connection._execute_clauseelement(self, multiparams, params)
  File "build/bdist.linux-x86_64/egg/sqlalchemy/engine/base.py", line 1060, in _execute_clauseelement
    compiled_sql, distilled_params
  File "build/bdist.linux-x86_64/egg/sqlalchemy/engine/base.py", line 1200, in _execute_context
    context)
  File "build/bdist.linux-x86_64/egg/sqlalchemy/engine/base.py", line 1413, in _handle_dbapi_exception
    exc_info
  File "build/bdist.linux-x86_64/egg/sqlalchemy/util/compat.py", line 203, in raise_from_cause
    reraise(type(exception), exception, tb=exc_tb, cause=cause)
  File "build/bdist.linux-x86_64/egg/sqlalchemy/engine/base.py", line 1193, in _execute_context
    context)
  File "build/bdist.linux-x86_64/egg/sqlalchemy/engine/default.py", line 508, in do_execute
    cursor.execute(statement, parameters)
  File "build/bdist.linux-x86_64/egg/MySQLdb/cursors.py", line 205, in execute
    self.errorhandler(self, exc, value)
  File "build/bdist.linux-x86_64/egg/MySQLdb/connections.py", line 36, in defaulterrorhandler
    raise errorclass, errorvalue
ProgrammingError: (_mysql_exceptions.ProgrammingError) (2014, "Commands out of sync; you can't run this command now") [SQL: u'SELECT `390528a5842828532c263eb2f0f2afcd_resources`.full_name AS `390528a5842828532c263eb2f0f2afcd_resources_full_name`, `390528a5842828532c263eb2f0f2afcd_resources`.type_name AS `390528a5842828532c263eb2f0f2afcd_resources_type_name`, `390528a5842828532c263eb2f0f2afcd_resources`.parent_type_name AS `390528a5842828532c263eb2f0f2afcd_resources_parent_type_name`, `390528a5842828532c263eb2f0f2afcd_resources`.name AS `390528a5842828532c263eb2f0f2afcd_resources_name`, `390528a5842828532c263eb2f0f2afcd_resources`.type AS `390528a5842828532c263eb2f0f2afcd_resources_type`, `390528a5842828532c263eb2f0f2afcd_resources`.policy_update_counter AS `390528a5842828532c263eb2f0f2afcd_resources_policy_update_counter`, `390528a5842828532c263eb2f0f2afcd_resources`.display_name AS `390528a5842828532c263eb2f0f2afcd_resources_display_name`, `390528a5842828532c263eb2f0f2afcd_resources`.email AS `390528a5842828532c263eb2f0f2afcd_resources_email`, `390528a5842828532c263eb2f0f2afcd_resources`.data AS `390528a5842828532c263eb2f0f2afcd_resources_data` \nFROM `390528a5842828532c263eb2f0f2afcd_resources` \nWHERE `390528a5842828532c263eb2f0f2afcd_resources`.type_name = %s'] [parameters: ('project/foo-test',)] (Background on this error at: http://sqlalche.me/e/f405)

### PR DESCRIPTION
**Pull request by blueandgold**
Fixes #1711, by enabling eager loading.

The KE scanner needs a different fix because it uses 2 explicit nested queries.  So, breaking them up into separate sequential queries.

### CODE REVIEW

File: google/cloud/forseti/scanner/scanners/ke_version_scanner.py [hunk 1]
```diff
@@ -119,17 +121,27 @@ def _retrieve(self):
...
                     session, 'kubernetes_cluster'):
-                service_config = list(data_access.scanner_iter(
-                    session, 'kubernetes_service_config',
-                    parent_type_name=ke_cluster.type_name))[0]
-
                 project_id = ke_cluster.parent.name
                 ke_clusters.append(
                     KeCluster.from_json(project_id,
-                                        service_config.data,
+                                        None,
```
Comment (joecheuk) [inline google/cloud/forseti/scanner/scanners/ke_version_scanner.py]: Was this field not used before?

Comment (blueandgold) [reply google/cloud/forseti/scanner/scanners/ke_version_scanner.py]: Correct, it was before.  I moved it into the 2nd query below, in line 143.

Comment (joecheuk) [reply google/cloud/forseti/scanner/scanners/ke_version_scanner.py]: Got it.

**Comment by blueandgold** [comment]
The travis failures are for lints.  Opened issue #1719 to fix separately.

**Pull request merged**